### ISSUE TITLE
Random Paragraphs added After Raw Html

### ISSUE THREAD

**Issue by LeoGerber** [jgm/pandoc#3717]
Given the following Markdown:

```
# lorem lorem lorum

Lorem ipsum dolor sit amet,  consetetur sadipscing elitr, sed diam nonumy <button>wat soll</button> eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed 
 diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. <button>dit</button>Stet clita kasd gubergren, no sea takimata
  sanctus est Lorem ipsum dolor sit amet.
```

And the following Parameters for translating from Markdown to HTML:
`pandoc.exe -s -o "somepath/output.html" "somepath/markdown.md"`

A paragraph (p tag) is inserted after the last button as can be seen below:
```
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <title></title>
  <style type="text/css">code{white-space: pre;}</style>
</head>
<body>
<h1 id="lorem-lorem-lorum">lorem lorem lorum</h1>
Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy
<button>
wat soll
</button>
eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
<button>
dit
</button>
<p>Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.</p>
</body>
</html>
```

This seems to be **unintended** and is pretty annoying. It breaks the layout of the document :-(

Pandoc Version: 1.19.2.1 -> Windows 7 64 bit

**Comment by LeoGerber** [comment]
@jgm Not even a comment on this?

**Comment by tarleb** [comment]
@LeoGerber The issue was closed by jgm via a commit that fixed the issue: 0f07404daf08156fac655f0ff907a6126c639450. The commit message contains more information on what was happening and how it's been fixed.  Thanks for reporting the bug!

**Comment by LeoGerber** [comment]
I was not even considering that this would be fixed that fast :-) Awesome! Thank you very much.@jgm
